cobc/cobc.c

```diff
@@ -4640,7 +4640,8 @@ process_filename (const char *filename)
 		fn->preprocess = cobc_main_strdup (output_name);
 	} else
 	if (save_all_src || save_temps
-	 || cb_compile_level == CB_LEVEL_PREPROCESS) {
+	 || (cb_compile_level == CB_LEVEL_PREPROCESS
+	 && !cb_depend_output_only)) {
```

Comment:
in this case please make that a single-line expression to have the indent at the || (or indent after its ( and on the next line &&,  but I think the one-liner is clearer to grasp here